### ISSUE TITLE
Insufficient color contrast for color input fields in UFIXIT for color contrast

### ISSUE THREAD

**Issue by robfentress** [ucfopen/UDOIT#161]
Just noticed that some of the feedback UFIXIT provides on failing the color contrast rule, can, itself be illegible because the hex value of the color used is displayed on top of a background of that color.  The hex value is displayed above the colored background as well, so this isn't that big a deal, but it is ironic and you may want to get rid of where the hex value text is shown over the background.
![screen shot 2016-05-31 at 3 03 07 pm](https://cloud.githubusercontent.com/assets/9356499/15687194/6935fb4c-2741-11e6-9298-26ab12c5cd86.png)

**Comment by bagofarms** [comment]
Good catch!

**Comment by techrangerfj** [comment]
@robfentress Hey, I was unable to reproduce this in the latest version of UDOIT, which version are you using?
![screen shot 2016-06-08 at 1 07 10 pm](https://cloud.githubusercontent.com/assets/7559106/15903587/6e1821fe-2d7b-11e6-8fef-cfe346dc56e5.png)

**Comment by robfentress** [comment]
Unfortunately, I am not the person who installed UDOIT, so I'm not sure,
but I know it is at least 2.0.4.  Was there a relevant change made between
these two versions?

On Wed, Jun 8, 2016 at 1:18 PM, Fenel Joseph <notifications@github.com>
wrote:

> @robfentress <https://github.com/robfentress> Hey, I was unable to
> reproduce this in the latest version of UDOIT, which version are you using?
> [image: screen shot 2016-06-08 at 1 07 10 pm]
> <https://cloud.githubusercontent.com/assets/7559106/15903587/6e1821fe-2d7b-11e6-8fef-cfe346dc56e5.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ucfcdl/UDOIT/issues/161#issuecomment-224663596>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AI7E03BcwgQg54tsQD1Q5AK42o56OE5Nks5qJvlTgaJpZM4Iq3Fq>
> .
>



-- 
Robert Fentress
Senior Accessibility Solutions Designer
540.231.1255

Technology-enhanced Learning & Online Strategies
Assistive Technologies
1180 Torgersen Hall
620 Drillfield Drive (0434)
Blacksburg, Virginia 24061

**Comment by robfentress** [comment]
2.0.5

> On Jun 8, 2016, at 2:12 PM, Robert Fentress <learn@vt.edu> wrote:
> 
> Unfortunately, I am not the person who installed UDOIT, so I'm not sure, but I know it is at least 2.0.4.  Was there a relevant change made between these two versions?
> 
> On Wed, Jun 8, 2016 at 1:18 PM, Fenel Joseph <notifications@github.com> wrote:
> @robfentress Hey, I was unable to reproduce this in the latest version of UDOIT, which version are you using?
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
> 
> -- 
> Robert Fentress
> Senior Accessibility Solutions Designer
> 540.231.1255
> 
> Technology-enhanced Learning & Online Strategies
> Assistive Technologies
> 1180 Torgersen Hall
> 620 Drillfield Drive (0434)
> Blacksburg, Virginia 24061

-- 
Bob Edmison
Director of Software Development, Learning Systems
Technology-enhanced Learning and Online Strategies 
Torgersen Hall 2220B, Virginia Tech (0292) 
620 Drillfield Drive 
Blacksburg VA. 24061 United States (EST)
Email: bedmison@vt.edu
Voice: 540.231.3957

**Comment by robfentress** [comment]
Does UDOIT inherit styles from any custom themes/branding applied at the
institution level?  Perhaps that is the issue.

On Wed, Jun 8, 2016 at 2:12 PM, Robert Fentress <learn@vt.edu> wrote:

> Unfortunately, I am not the person who installed UDOIT, so I'm not sure,
> but I know it is at least 2.0.4.  Was there a relevant change made between
> these two versions?
>
> On Wed, Jun 8, 2016 at 1:18 PM, Fenel Joseph <notifications@github.com>
> wrote:
>
>> @robfentress <https://github.com/robfentress> Hey, I was unable to
>> reproduce this in the latest version of UDOIT, which version are you using?
>> [image: screen shot 2016-06-08 at 1 07 10 pm]
>> <https://cloud.githubusercontent.com/assets/7559106/15903587/6e1821fe-2d7b-11e6-8fef-cfe346dc56e5.png>
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ucfcdl/UDOIT/issues/161#issuecomment-224663596>, or mute
>> the thread
>> <https://github.com/notifications/unsubscribe/AI7E03BcwgQg54tsQD1Q5AK42o56OE5Nks5qJvlTgaJpZM4Iq3Fq>
>> .
>>
>
>
>
> --
> Robert Fentress
> Senior Accessibility Solutions Designer
> 540.231.1255
>
> Technology-enhanced Learning & Online Strategies
> Assistive Technologies
> 1180 Torgersen Hall
> 620 Drillfield Drive (0434)
> Blacksburg, Virginia 24061
>



-- 
Robert Fentress
Senior Accessibility Solutions Designer
540.231.1255

Technology-enhanced Learning & Online Strategies
Assistive Technologies
1180 Torgersen Hall
620 Drillfield Drive (0434)
Blacksburg, Virginia 24061

**Comment by techrangerfj** [comment]
@robfentress I'll investigate. If I can't find where it's done, I'll add the logic necessary to produce the same experience in all cases.

**Comment by robfentress** [comment]
Oh, yeah, I just realized this is a bigger deal than I thought, since you can't just get rid of that text, because that is where you enter in the new hex values.  I hadn't actually tried to fix the colors, so I didn't really understand how that worked.  

I'm not sure if this should be a different ticket, but it wasn't at all clear to me that these were fields where you could actually enter text.  I just thought you clicked on the chits in the Foreground Color Palette to set the colors.  Might be good to provide a little more guidance how that is supposed to work.

**Comment by bagofarms** [comment]
I renamed this issue to make it a little clearer.  We have changed the code to automatically choose white or black based on the color chosen, but I think the interface could be done better.  We have a few options here:

- As soon as you start typing in the box, make the background of the field white.
- Change the color input to be a color box with an input field next to it.  You can click the color box to select from a palette or input the text manually in the input box.  The input box would automatically update with the palette choice, and the palette box would update automatically with the text input value.

I'll need to run this by the team here, but i would appreciate input from others.  Maybe @robfentress @cooperfellows @iturgeon @rebelaide

**Comment by robfentress** [comment]
Working on some other stuff right now, but will get explore/give some thought to this when I get a chance.

**Comment by cooperfellows** [comment]
@bagofarms Is this sort of what you mean for your second suggestion:

 - ![image](https://cloud.githubusercontent.com/assets/6798541/22303993/a128418a-e303-11e6-85a8-6f52faa7f7a1.png)

I'm not advocating for another library, just trying to make sure I understand the suggestion, the above design example [came from here](https://itsjavi.com/bootstrap-colorpicker/).

I think we could mimic that behavior pretty easily with our current ColorPicker library.

The first suggestion would be tough, at what point would you update the background color again? Or are you saying that the text input would remain white?

**Comment by emilysachs** [comment]
Partially addressed in https://github.com/ucfopen/UDOIT/issues/280 by making hex text switch between black and white based on darkness of background color

**Comment by bagofarms** [comment]
This UI is going to change completely in 3.0.0, so I'm going to close this issue.
